src/Formats/FormatFactory.cpp

```diff
@@ -232,14 +233,18 @@ InputFormatPtr FormatFactory::getInput(
     const Settings & settings = context->getSettingsRef();
     const auto & file_segmentation_engine = getCreators(name).file_segmentation_engine;
 
+    size_t max_threads = settings.max_threads;
+    if (number_of_streams.has_value() && number_of_streams.value())
```

Comment:
if we anyway skip case with number_of_streams = 0 without any exception but as noop, so maybe get rid of std::optional and use 0 as default and noop? just to simplify